content/local-setup.md

```diff
@@ -0,0 +1,283 @@
+```bash
+solana-keygen new --outfile ~/<FILE_PATH>
+```
+
+To update the `Keypair Path` for the Solana CLI config, using the `solana config set --keypair` command and specify the path of the new keypair (Solana CLI will now use the keypair from the specified path when running commands).
+
+```bash
+solana config set --keypair ~/<FILE_PATH>
+```
+
+To view the `publickey` of the currently configured keypair, run the `solana address` command.
+
+```bash
+solana address
+```
+
+To view the SOL balance of the of the currently configured keypair, run the `solana balance` command.
+
+```bash
+solana balance
+```
+
+To airdrop SOL for testing (on Devnet or Localhost), run the `solana airdrop <AMOUNT>` command (Note that while on Devnet you are limited to 2 SOL per airdrop).
+
+```bash
+solana airdrop 2
+```
+
+While deploying and testing on Localhost, you must run a test validator using the `solana-test-validator` command.
+
+```bash
+solana-test-validator
+```
+
+While on Localhost, you can view program logs by running the `solana logs` command in a separate terminal.
+
+```bash
+solana logs
+```
+
+To view program logs while on Devnet, use the `solana logs | grep "<PROGRAM_ID> invoke"` command and specify the program Id. If the program Id is not specified, then the `solana logs` command will return a constant stream of logs from Devnet. You can then specify the number of lines (of program logs) to return by including `-A <NUMBER_OF_LINES_TO_RETURN>` after the command.
+
+```bash
+solana logs | grep "<PROGRAM_ID> invoke" -A <NUMBER_OF_LINES_TO_RETURN>
+```
+
+To deploy a program, you using the `solana program deploy` command and specify the file path of the program. The command to deploy a program along with the file path is provided as an output when you build the program. We’ll go over this shortly in the demo.
+
+```bash
+solana program deploy <PATH>
+```
+
+Lastly, you can view the details for Solana CLI commands using the `solana --help` command.
+
+```bash
+solana --help
+```
+
+## Solana Program from Local Environment
+
+To create a new Rust project run the following in a new terminal.
+
+```bash
+cargo new --lib <PROJECT_FILE_NAME>
+```
+
+Open the new project and update the `Cargo.toml` file. Remember to specify the latest version of `solana-program`.
+
+```rust
+[package]
+name = "<PROJECT_FILE_NAME>"
+version = "0.1.0"
+edition = "2021"
+
+[features]
+no-entrypoint = []
+
+[dependencies]
+solana-program = "1.10.7"
+
+[lib]
+crate-type = ["cdylib", "lib"]
+```
+
+To build a Solana program in Rust, run the `cargo build-bpf` command. The output provides the `solana program deploy` command specifying the file path to deploy the program.
+
+```bash
+cargo build-bpf
+```
+
+Finally, to deploy the program after building, run the `solana program deploy` command output from `cargo build-bpf`.
+
+```bash
+solana program deploy <PATH>
+```
+
+# Demo
+
+Let’s practice by building and deploying the “Hello World!” program from lesson one of this module.
```

Comment:
This sentence is wrong too - "Hello World" was lesson one of the last module. I would just link to it instead:
> Let's practice by building and deploying  the "Hello World!" program that we created in the [Hello World lesson](./content/hello-world-program.md). We'll do this all locally, including deploying to a local test validator.